rmpv/src/ext/de.rs

```diff
@@ -508,17 +508,7 @@ impl<'de> Deserializer<'de> for &'de ValueRef<'de> {
     fn deserialize_newtype_struct<V>(self, _name: &'static str, visitor: V) -> Result<V::Value, Self::Error>
         where V: Visitor<'de>
     {
-        match self {
-            &ValueRef::Array(ref v) => {
-                let iter = v.into_iter();
-                if iter.len() != 1 {
-                    return Err(de::Error::invalid_length(iter.len(), &"array with one element"));
-                }
-
-                visitor.visit_seq(SeqDeserializer::new(iter))
-            }
-            other => Err(de::Error::invalid_type(other.unexpected(), &"array")),
-        }
+        visitor.visit_newtype_struct(self)
```

Comment:
I guess it would introduce some ambiguity and make it difficult to work with nested types. Handling of something like `struct A(Vec<i32>)` would require some code to decode it correctly.
Is it really necessary?